src/extensions/extension-loader.ts

```diff
@@ -248,25 +247,7 @@ export class ExtensionLoader extends Singleton {
   }
 
   loadOnMain() {
-    registries.MenuRegistry.createInstance();
-
-    logger.debug(`${logModule}: load on main`);
-    this.autoInitExtensions(async (extension: LensMainExtension) => {
-      // Each .add returns a function to remove the item
-      const removeItems = [
-        registries.MenuRegistry.getInstance().add(extension.appMenus),
-      ];
-
-      this.events.on("remove", (removedExtension: LensRendererExtension) => {
-        if (removedExtension.id === extension.id) {
-          removeItems.forEach(remove => {
-            remove();
-          });
-        }
-      });
-
-      return removeItems;
-    });
+    this.autoInitExtensions(() => Promise.resolve([]));
```

Comment:
Notice how we no longer need to dispose items from a state when extension is disabled or removed.